pkg/kademlia/routing.go

```diff
@@ -149,6 +149,19 @@ func (rt *RoutingTable) GetBucketIds() (storage.Keys, error) {
 	return kbuckets, nil
 }
 
+// DumpNodes iterates through all nodes in the nodeBucketDB and marshals them to &pb.Nodes, then returns them
+func (rt *RoutingTable) DumpNodes() ([]*pb.Node, error) {
+	var nodes []*pb.Node
+	err := rt.iterate(storj.NodeID{}, func(newID storj.NodeID, protoNode []byte) error {
+		newNode := pb.Node{}
+		err := proto.Unmarshal(protoNode, &newNode)
+		nodes = append(nodes, &newNode)
```

Comment:
Should we append when err != nil?

Reply:
Good point. I'll rework this to use ErrorGroup handling.